server/src/main/java/io/druid/client/CachingClusteredClient.java

```diff
@@ -443,21 +443,25 @@ public void run()
                                   public void run()
                                   {
                                     if (cachePopulator != null) {
-                                      try {
-                                        Futures.allAsList(cacheFutures).get();
-                                        cachePopulator.populate(cacheData);
-                                        // Help out GC by making sure all references are gone
-                                        cacheFutures.clear();
-                                        cacheData.clear();
-                                      }
-                                      catch (Exception e) {
-                                        log.error(e, "Error populating cache");
-                                        throw Throwables.propagate(e);
-                                      }
+                                      ListenableFuture<?> future = Futures.allAsList(cacheFutures);
+                                      future.addListener(
```

Comment:
future.addListener will be called even if the future is cancelled or exceptions out.
It might be safer to use Futures.addCallback and put the code within the onSuccess method.